### PR TITLE
Add missing STJ xml docs

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Companion PR to https://github.com/dotnet/dotnet-api-docs/pull/9297

### CODE REVIEW

File: src/libraries/System.Text.Json/src/CompatibilitySuppressions.xml [hunk 1]
```diff
@@ -57,6 +57,54 @@
     <Right>lib/netstandard2.0/System.Text.Json.dll</Right>
     <IsBaselineSuppression>true</IsBaselineSuppression>
   </Suppression>
+  <Suppression>
```
Comment (eiriktsarpalis) [inline src/libraries/System.Text.Json/src/CompatibilitySuppressions.xml:60]: Explicitly adding default constructors tripped off the API compat checker necessitating these suppressions. FYI @carlossanlop @ViktorHofer

File: src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.cs [hunk 2]
```diff
@@ -113,9 +113,10 @@ internal int GetElementIndex(JsonNode? node)
         }
 
         /// <summary>
-        /// Returns enumerator that wraps calls to <see cref="JsonNode.GetValue{T}"/>.
+        /// Returns an enumerable that wraps calls to <see cref="JsonNode.GetValue{T}"/>.
```
Comment (carlossanlop) [inline src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.cs:116]: This change didn't get reflected in the dotnet-api-docs PR, FYI. APIs with existing docs don't get updated by the porting tool.

**Review by carlossanlop — APPROVED** [review]
LGTM. Left a node, and also there were some suggestions in the dotnet-api-docs PR in case you want to modify them here too.

File: src/libraries/System.Text.Json/src/CompatibilitySuppressions.xml (hunk 1 above)
Comment (ViktorHofer) [reply src/libraries/System.Text.Json/src/CompatibilitySuppressions.xml:60]: That needs to be investigated. @eiriktsarpalis do you still have the error message available? I wonder if APICompat complains about the member being on the ref but not on the src or the other way around.

cc @ericstj as I'm out the next two weeks

Comment (ViktorHofer) [reply src/libraries/System.Text.Json/src/CompatibilitySuppressions.xml:60]: Actually, I just looked at the reference source and the type defines an explicit constructor with parameters, so adding a default one is an API addition that you probably don't want to make. Or do you?

**Review by ViktorHofer — CHANGES_REQUESTED** [review]
Marking as no-merge until we resolve the public API discussion

File: src/libraries/System.Text.Json/src/CompatibilitySuppressions.xml (hunk 1 above)
Comment (ericstj) [reply src/libraries/System.Text.Json/src/CompatibilitySuppressions.xml:60]: Yeah these are all ref/lib comparisons.  

> Explicitly adding default constructors

This isn't a case of explicit vs implicit.  Implicit default constructors [only exist if no other constructors are defined](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/instance-constructors#parameterless-constructors). In these cases you're adding a default constructor to types that didn't have an implicit one.  This a real error. 
image

If the API docs tooling is telling you this constructor exists -- it is mistaken.  Don't add this API.

Comment (ViktorHofer) [reply src/libraries/System.Text.Json/src/CompatibilitySuppressions.xml]: > Yeah these are all ref/lib comparisons.

That's correct. I was asking about the error message as it provides information about if the error applies to the right or the left side. As this comparison is done in strict mode, the CP0002 error could apply to either missing API on the left or on the right side. Unfortunately, the XML doesn't provide that detail.

That said, in this case it's clear that this is about an API addition on the right side.

Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/src/CompatibilitySuppressions.xml]: Thanks. The API docs tooling is actually correct -- it was complaining about a missing parameter that I mistook for a missing default constructor (similar to a few other types in the undoc list). I just reverted the unnecessary constructors and suppressions.

File: src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.cs (hunk 2 above)
Comment (eiriktsarpalis) [reply src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.cs:116]: Fixed.

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**